src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/AutoInsert/RemoteAutoInsertService.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor.Cohost.Handlers;
+using Microsoft.CodeAnalysis.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.DocumentMapping;
+using Microsoft.CodeAnalysis.Razor.Formatting;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Protocol.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.CodeAnalysis.Remote.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.LanguageServer.Protocol;
+using Response = Microsoft.CodeAnalysis.Razor.Remote.RemoteResponse<Microsoft.CodeAnalysis.Razor.Protocol.AutoInsert.RemoteInsertTextEdit?>;
+using RoslynFormattingOptions = Roslyn.LanguageServer.Protocol.FormattingOptions;
+using RoslynInsertTextFormat = Roslyn.LanguageServer.Protocol.InsertTextFormat;
+using VsLspFormattingOptions = Microsoft.VisualStudio.LanguageServer.Protocol.FormattingOptions;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor;
+
+internal class RemoteAutoInsertService(in ServiceArgs args)
+    : RazorDocumentServiceBase(in args), IRemoteAutoInsertService
+{
+    internal sealed class Factory : FactoryBase<IRemoteAutoInsertService>
+    {
+        protected override IRemoteAutoInsertService CreateService(in ServiceArgs args)
+            => new RemoteAutoInsertService(in args);
+    }
+
+    private readonly IAutoInsertService _autoInsertService
+        = args.ExportProvider.GetExportedValue<IAutoInsertService>();
+    private readonly IDocumentMappingService _documentMappingService
+        = args.ExportProvider.GetExportedValue<IDocumentMappingService>();
+    private readonly IFilePathService _filePathService =
+        args.ExportProvider.GetExportedValue<IFilePathService>();
+    private readonly IRazorFormattingService _razorFormattingService =
+        args.ExportProvider.GetExportedValue<IRazorFormattingService>();
+    private readonly ILogger _logger =
```

Comment:
It's unnecessary to create your own logger for a remote service. An `ILogger` is created by each `RazorBrokeredServiceBase` and is accessible via a protected `Logger` field.

Reply:
Didn't realize it, thanks, removing this one.